File: src/minInRotatedSortedArray.java
// Time Complexity: log(n);
// Space Complexity: o(1);
// Did it run on leetcode: Yes
// Problems faced: No
public class minInRotatedSortedArray {

    public int findMin(int[] nums) {

        if(nums.length == 1){
            return nums[0];
        }

        int low = 0;
        int high = nums.length -1;

        while ( low < high ){

            int mid = low + (high -low)/2;

            if(nums[low] <= nums[high]){
                return nums[low];
            }

            if( (mid ==0 || nums[mid] < nums[mid-1] ) && nums[mid] < nums[mid+1] ){
                return nums[mid];
            }else if( nums[mid] > nums[high] ){
                low = mid+1;
            }else{
                high = mid-1;
            }

        }

        return nums[low];

    }

}

File: MinInRotatedSortedArray.js
// Time Complexity : O(logn)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes

// Your code here along with comments explaining your approach

/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
  if (nums === null || nums.length === 0) {
    return -1;
  }
  var low = 0;
  var high = nums.length - 1;
  var min = nums[low];

  while (low <= high) {
    var mid = Math.floor((high + low) / 2);
    //Traverse left side
    if (nums[low] <= nums[mid]) {
      if (nums[low] < min) {
        min = nums[low];
      }
      low = mid + 1;
    } else {
      if (nums[mid] < min) {
        min = nums[mid];
      }
      high = mid - 1;
    }
  }
  return min;
};
